spec/unit/provider/oneview_enclosure_group_synergy_spec.rb

```diff
@@ -0,0 +1,126 @@
+# limitations under the License.
+################################################################################
+
+require 'spec_helper'
+
+provider_class = Puppet::Type.type(:oneview_enclosure_group).provider(:synergy)
+api_version = login[:api_version] || 200
+resourcetype = OneviewSDK.resource_named(:EnclosureGroup, api_version, 'Synergy')
+
+describe provider_class, unit: true do
+  include_context 'shared context'
+
+  context 'given the create parameters' do
+    let(:resource) do
+      Puppet::Type.type(:oneview_enclosure_group).new(
+        name: 'EnclosureGroup',
+        ensure: 'present',
+        data:
+          {
+            'name'                         => 'Enclosure Group',
+            'interconnectBayMappingCount'  => '8',
+            'stackingMode'                 => 'Enclosure',
+            'interconnectBayMappings'      =>
+            [
+              {
+                'interconnectBay' => '1',
+                'logicalInterconnectGroupUri' => nil
+              },
+              {
+                'interconnectBay' => '2',
+                'logicalInterconnectGroupUri' => nil
+              },
+              {
+                'interconnectBay' => '3',
+                'logicalInterconnectGroupUri' => nil
+              },
+              {
+                'interconnectBay' => '4',
+                'logicalInterconnectGroupUri' => nil
+              },
+              {
+                'interconnectBay' => '5',
+                'logicalInterconnectGroupUri' => nil
+              },
+              {
+                'interconnectBay' => '6',
+                'logicalInterconnectGroupUri' => nil
+              },
+              {
+                'interconnectBay' => '7',
+                'logicalInterconnectGroupUri' => nil
+              },
+              {
+                'interconnectBay' => '8',
+                'logicalInterconnectGroupUri' => nil
+              }
+            ]
+          },
+        provider: 'synergy'
+      )
+    end
+
+    let(:provider) { resource.provider }
+
+    let(:instance) { provider.class.instances.first }
+
+    it 'should be an instance of the provider Ruby' do
+      expect(provider).to be_an_instance_of Puppet::Type.type(:oneview_enclosure_group).provider(:synergy)
+    end
+
+    it 'should be able to find the resource' do
+      test = resourcetype.new(@client, resource['data'])
+      allow(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([test])
+      provider.exists?
+      expect(provider.found).to be
+    end
+
+    it 'runs through the create method' do
+      allow(resourcetype).to receive(:find_by).and_return([])
+      test = resourcetype.new(@client, resource['data'])
+      allow_any_instance_of(resourcetype).to receive(:create).and_return(test)
+      provider.exists?
+      expect(provider.create).to be
+    end
+
+    it 'deletes the resource' do
+      resource['data']['uri'] = '/rest/fake'
```

Comment:
If I remove this line:

```ruby
it 'deletes the resource' do
      resource['data']['uri'] = '/rest/fake'
      test = resourcetype.new(@client, resource['data'])
      allow(resourcetype).to receive(:find_by).with(anything, resource['data']).and_return([test])
      allow(resourcetype).to receive(:find_by).with(anything, 'name' => resource['data']['name']).and_return([test])
      expect_any_instance_of(OneviewSDK::Client).to receive(:rest_delete).and_return(FakeResponse.new('uri' => '/rest/fake'))
      provider.exists?
      expect(provider.destroy).to be
end
```

I got this error:

1) Puppet::Type::Oneview_datacenter::ProviderSynergy given the min parameters deletes the resource
     Failure/Error: datacenter.map(&:remove)
     
     OneviewSDK::IncompleteResource:
       Please set uri attribute before interacting with this resource
     ./lib/puppet/provider/oneview_datacenter/c7000.rb:37:in `map'
     ./lib/puppet/provider/oneview_datacenter/c7000.rb:37:in `destroy'
     ./spec/unit/provider/oneview_datacenter_synergy_spec.rb:113:in `block (3 levels) in <top (required)>'